netty/src/test/java/io/grpc/netty/NettyClientTransportTest.java

```diff
@@ -571,14 +573,16 @@ public void onReady() {
       this.method = method;
       this.headers = headers;
       stream.writeHeaders(new Metadata());
-      stream.request(1);
```

Comment:
When deframing in the client thread (later commit, so this could be moved out of this commit entirely), calls to `request` trigger a call to `messagesAvailable` on the listener. Calling `request(1)` here in the `EchoServerStreamListener` constructor causes a null pointer exception, because the stream doesn't have a listener set until line 615, after the `EchoServerStreamListener` is created and set on the stream. This can be avoided by doing a null check in `AbstractStream`, but it seemed cleaner just not to invoke these methods when the listener isn't set.

But I realize now that this either needs to be documented on the `{Client,Server,}Stream` API and/or still do the a null check in `AbstractStream.messagesAvailable`